io/net/port_bounce.py

```diff
@@ -0,0 +1,156 @@
+        self.count = int(self.params.get("count", '*', default="2"))
+        self.prompt = ">"
+
+    def fc_login(self, ip, username, password):
+        '''
+        telnet Login method for remote fc switch
+        '''
+        self.tnc = telnetlib.Telnet(ip)
+        self.tnc.read_until('login: ')
+        self.tnc.write(username + '\n')
+        self.tnc.read_until('assword: ')
+        self.tnc.write(password + '\n')
+        ret = self.tnc.read_until(self.prompt)
+        assert self.prompt in ret
+
+    def _send_only_result(self, command, response):
+        output = response.splitlines()
+        if command in output[0]:
+            output.pop(0)
+        output.pop()
+        output = [element.lstrip()+'\n' for element in output]
+        response = ''.join(output)
+        response = response.strip()
+        return ''.join(response)
+
+    def fc_run_command(self, command, timeout=300):
+        '''
+        Telnet Run command method for running commands on fc switch
+        '''
+        self.log.info("Running the command on fc swicth %s", command)
+        self.tnc.write(command + '\n')
+        response = self.tnc.read_until(self.prompt)
+        return self._send_only_result(command, response)
+
+    def test(self):
+        if self.fc_type == "fcoe":
+            if self.fcoe_fc == "yes":
+                test = self.portdisable_fcoe_fc
+            else:
+                test = self.portdisable_fcoe
+        else:
+            test = self.portdisable
+        self.port_bounce(test)
+
+    def port_bounce(self, test):
+        for port in self.port_ids:
+            self.log.info("Portbounce test for port %s for %s times",
+                          port, self.count)
+            for i in range(self.count+1):
+                self.log.info("Port %s, Short Portbounce test", port)
+                test(self, port, SHORT_BOUNCE_TIME)
+                self.log.info("Port %s, Long Portbounce test", port)
+                test(self, port, LONG_BOUNCE_TIME)
+                self.log.info("%s portdisable/portenable tests completed for \
+                              Port %s", i+1, port)
+
+    @staticmethod
+    def portdisable(self, port, sleep_time):
+        self.fc_login(self.switch_name, self.userid, self.password)
+        switch_info = self.fc_run_command("switchshow")
+        self.log.info("Swicth info: %s", switch_info)
+        self.log.info("Disable port %s", port)
+        try:
+            self.fc_run_command("portdisable %s" % port)
+        except CommandFailed as cf:
+            self.log.info("port disable failed for port %s, details: %s",
+                          port, str(cf))
+        time.sleep(sleep_time)
+        self.log.info("Enable port %s", port)
+        try:
+            self.fc_run_command("portenable %s" % port)
+        except CommandFailed as cf:
+            self.log.info("port enable failed for port %s, details: %s",
+                          port, str(cf))
```

Comment:
Also we expect 'remote port timeout' messages  or 'No light siganl' at linux side, can we check that ?